app/controllers/concerns/flowable.rb

```diff
@@ -22,6 +22,10 @@ def step_prefix
     delegate :forward_path, to: :flow_service
     helper_method :forward_path, :journey_type
 
+    def flow_param(flow_param)
+      @flow_param = flow_param
+    end
```

Comment:
I wonder if rather than relying on two calls to the flow service, the flow params could be passed into `go_forward`. I think it requires a little refactoring to ensure that memoized objects aren't reused out of context by other processes. So ...

First step remove the line:
```ruby
delegate :forward_path, to: :flow_service
```
Then refactor the code:
```ruby

    def go_forward(flow_param=nil)
      forward_path = flow_service(flow_params).forward_path
      if path?(forward_path)
        redirect_to forward_path
      else
        render plain: forward_path
      end
    end

    def flow_service(flow_params=nil)
      Flow::BaseFlowService.flow_service_for(
        journey_type,
        legal_aid_application: legal_aid_application,
        current_step: current_step,
        params: flow_param
      )
    end
```
Then in the controller rather than:
```ruby
flow_param(dependant)
go_forward
```
We use:
```ruby
go_forward(dependant)
```